### PR TITLE
DEV 83 1/2: Update Footer to the new Design

### PR DESCRIPTION
**Pull request by ApplePieAngel**
**Merge notes: Will need to be merged with #73 at least. (Can ignore AppBar)**

# Description
The purpose of this change is to update the footer component to align with the new design created by the design team.

Reasons for changes: 
- Non external links have been changed to useHistory in order to maintain userHistory when navigating pages
- Grid and CSS has been added to centralize the font/make the footer more responsive
- Donate doesn't actually redirect to any page yet, but was added as its displayed in the Figma design.
**(12/28/2022 Update)**
I've added unit test cases covering the links and copyright string to this PR.
- MUI palette colors doesn't play nice with unit tests. They always render as undefined and error out the tests. So I've converted it back to hardcoded values.

The reasoning behind my test cases styling is inspired from "Mastering React Test-Driven Development" by Daniel Irvine.

Related to DEV-83

## Screenshots
**Before**

chrome_2bGK1lIYdu

chrome_3P8anlP15J

**After**

chrome_9tXUwiP911

chrome_vSeCdDjXyc

## Type of change
Design Change

# How Has This Been Tested?
Using Chrome Browser + Chrome Developer Tools to test various breakpoints. 
As well as manually clicking through links. 
**Added and tested with unit tests to cover the components.**

### CODE REVIEW

File: src/components/Footer/Footer.jsx [hunk 1]
```diff
@@ -26,88 +30,96 @@ const styles = (theme) => ({
...
+            className={classes.buttonLink}
+          >
+            Community Guidelines
+          </Button>
+        </Grid>
+        <Grid item>
+          <Button
+            className={classes.buttonLink}
+            onClick={() => {
+              history.push({
```
Comment (deepakvenkat) [inline src/components/Footer/Footer.jsx:77]: Why do we need this instead of just a `href` ?

Comment (ApplePieAngel) [reply src/components/Footer/Footer.jsx:77]: Href elements would trigger a page refresh which would reset the application states. In this instance it means that the application would forget if the user was logged in or not which is relevant for other components and single page applications. 

If not useHistory, we can also use NavLink or Link which is part of React Router.

**Comment by ApplePieAngel** [comment]
As an update, I've updated all the Buttons to NavLinks. I've also added underlines under external links as I felt it wasn't clear to users if they were links previously.

Code_wlVp4SOy0w
Code_afeQj0rZzF

**Comment by ApplePieAngel** [comment]
As another update, I know several months ago we talked about not liking how testing was currently done. I've done a lot of research (books, conference talks, articles, etc) and added unit tests for this PR as my recommendation and my proposal for how unit tests should look like going forward.

**Review by InManuBytes — APPROVED** [review]
LGTM

**Pull request merged**